src/stdlib_selection.fypp

```diff
@@ -86,6 +86,7 @@ contains
           if(present(right)) r = right
 
           if(k < 1_ip .or. k > size(a, kind=ip) .or. l > r .or. l < 1_ip .or. &
+              k < l .or. k > r .or.      &             !i.e. if k is not in the interval [l; r]  
```

Comment:
Good catch.

Reply:
Actually with these changes, I think we can also remove the criteria `k < 1_ip .or. k > size(a, kind=ip)`

Reply:
good point. Indeed, they can be removed.

Reply:
Done